sdk/python/feast/infra/offline_stores/bigquery.py

```diff
@@ -225,7 +231,8 @@ def _block_until_done():
 
         today = date.today().strftime("%Y%m%d")
         rand_id = str(uuid.uuid4())[:7]
-        path = f"{self.client.project}.{self.config.offline_store.dataset}.historical_{today}_{rand_id}"
+        dataset_project = self.config.offline_store.project or self.client.project
```

Comment:
Under what circumstances would `self.client.project` be different from `self.config.offline_store.project`